portal-ui/src/screens/Console/Tenants/TenantDetails/EditTenantMonitoringModal.tsx

```diff
@@ -0,0 +1,207 @@
+  labels,
+  annotations,
+  nodeSelector,
+  diskCapacityGB,
+  serviceAccountName,
+  tenantName,
+  tenantNamespace,
+}: IEditTenantMonitoringProps) => {
+  const [validationErrors, setValidationErrors] = useState<any>({});
+  const [newLabels, setNewLabels] = useState<IKeyValue[]>(
+    labels.length > 0 ? [...labels] : [{ key: "", value: "" }]
+  );
+  const [newAnnotations, setNewAnnotations] = useState<IKeyValue[]>(
+    annotations.length > 0 ? [...annotations] : [{ key: "", value: "" }]
+  );
+  const [newNodeSelector, setNewNodeSelector] = useState<IKeyValue[]>(
+    nodeSelector.length > 0 ? [...nodeSelector] : [{ key: "", value: "" }]
+  );
+  const [newImage, setNewImage] = useState<string>(image);
+  const [newDiskCapacityGB, setNewDiskCapacityGB] =
+    useState<number>(diskCapacityGB);
+  const [newServiceAccountName, setNewServiceAccountName] =
+    useState<string>(serviceAccountName);
+
+  const [labelsError, setLabelsError] = useState<any>({});
+  const [annotationsError, setAnnotationsError] = useState<any>({});
+  const [nodeSelectorError, setNodeSelectorError] = useState<any>({});
+
+  const trim = (x: IKeyValue[]): IKeyValue[] => {
+    let retval: IKeyValue[] = [];
+    for (let i = 0; i < x.length; i++) {
+      if (x[i].key !== "") {
+        retval.push(x[i]);
+      }
+    }
+    return retval;
+  };
+  const submitMonitoringInfo = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    api
+      .invoke(
+        "PUT",
+        `/api/v1/namespaces/${tenantNamespace}/tenants/${tenantName}/monitoring`,
+        {
+          labels: trim(newLabels),
+          annotations: trim(newAnnotations),
+          nodeSelector: trim(newNodeSelector),
+          image: newImage,
+          diskCapacityGB: newDiskCapacityGB,
+          serviceAccountName: newServiceAccountName,
+        }
+      )
+      .then(() => {
+        onClose(true);
+      })
+      .catch((err: ErrorResponseHandler) => {});
+  };
+
+  return (
+    <ModalWrapper
+      onClose={() => onClose(true)}
+      modalOpen={open}
+      title="Edit Prometheus Monitoring Configuration"
+    >
+      <form noValidate autoComplete="off" onSubmit={submitMonitoringInfo}>
+        <h4>Image</h4>
+        <InputBoxWrapper
+          id={`image`}
+          label={""}
+          placeholder={"Image"}
+          name={`image`}
+          value={newImage}
+          onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+            setNewImage(event.target.value);
+          }}
+          key={`image`}
+          error={validationErrors[`image`] || ""}
+        />
+        <h4>Disk Capacity (GB)</h4>
+        <InputBoxWrapper
+          id={`diskCapacityGB`}
+          label={""}
+          placeholder={"Disk Capacity (GB)"}
+          name={`diskCapacityGB`}
+          value={newDiskCapacityGB as any as string}
+          onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+            setNewDiskCapacityGB(event.target.value as any as number);
```

Comment:
This can be avoided by converting the value to an integer:
```suggestion
            setNewDiskCapacityGB(parseInt(event.target.value));
```

Reply:
Another option would be to set `newDiskCapacityGB` state as a string